flake.nix

```diff
@@ -1,61 +1,36 @@
 {
   description = "nix.dev static website";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.05";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.11";
   inputs.flake-utils.url = "github:numtide/flake-utils/master";
-  inputs.poetry2nix = {
-    inputs.flake-utils.follows = "flake-utils";
-    inputs.nixpkgs.follows = "nixpkgs";
-    url = "github:nix-community/poetry2nix/master";
-  };
 
-  outputs = { self, nixpkgs, flake-utils, poetry2nix }:
+  outputs = { self, nixpkgs, flake-utils}:
     flake-utils.lib.eachDefaultSystem (system:
       let
         pkgs = import nixpkgs {
           inherit system;
-          overlays = [ poetry2nix.overlay ];
         };
-        poetryOverrides = self: super: {
-
-          # Refs https://github.com/nix-community/poetry2nix/issues/218#issuecomment-981615612
-          typing-extensions = super.typing-extensions.overridePythonAttrs (
-            old: {
-              buildInputs = (old.buildInputs or [ ]) ++ [ self.flit-core ];
-            }
-          );
-
-          sphinx-design = super.sphinx-design.overridePythonAttrs (
-            old: {
-              buildInputs = (old.buildInputs or [ ]) ++ [ self.flit-core ];
-            }
-          );
+      in {
+        defaultPackage = pkgs.stdenv.mkDerivation {
+          name = "nix-dev";
+          src = self;
+          buildInputs = with pkgs.python310Packages; [
+            livereload
+            myst-parser
+            sphinx
+            sphinx-book-theme
```

Comment:
Yes, something so seemingly simple as changing the version of a package is quite involved and requires a good understanding of the Nix language. That's because Nixpkgs' notion of a package currently is "function from packages (i.e., dependencies) to a (Nix language level) derivation with additional attributes". There is no way to declare dependencies and parameters, such as the version, distinctly in these function arguments, and some packages lump them together, some only allow passing dependencies. See this [Discourse thread for a discussion](https://discourse.nixos.org/t/proposal-move-values-from-derivation-attributes-to-function-arguments/20697/8).

`override` lets you re-evaluate the package function with new arguments (too coarse), `overrideAttrs` let's you re-evaluate the inner `mkDerivation` call with new arguments (too involved: fields were traditionally interdependent using `rec`, compounded by source fetching often needing extra infrastructure). Only somewhat recently we got a new interface thanks to @roberth that allows more granular use of `overrideAttrs`, but that's still dark magic underneath and by far not as simple for the user as setting configuration parameters on NixOS modules.

I have high hopes in the Nixpkgs architecture team around @infinisil to develop a more consistent, convenient, and easier to understand package paradigm, such as modules.

Reply:
Python packages should always use `.override` and/or `.overridePythonAttrs`. If you use `.overrideAttrs` or `.overrideDerivation` instead of `.overridePythonAttrs` you go one respectively two layers "too deep" bypassing the `buildPython*` and/or `stdenv.mkDerivation` routines.

The Python section in the Nixpkgs manual gives plenty of examples on how to override a package.